### PR TITLE
Fix for detecting cores

### PR DESCRIPTION
**Pull request by PavelBal**

### CODE REVIEW

File: R/ospsuite-env.R [hunk 1]
```diff
@@ -20,7 +20,7 @@ ospsuiteEnv$formatNumericsDigits <- 5
 ospsuiteEnv$formatNumericsSmall <- 2
 
 # Number of cores to use for simualtions and sensitivity. Default to number of cores on the machine - 1
-ospsuiteEnv$numberOfCores <- parallel::detectCores() - 1
+ospsuiteEnv$numberOfCores <- function(){parallel::detectCores() - 1}
```
Comment (PavelBal) [inline R/ospsuite-env.R:23]: I think the value was assigned upon building the package. It has to be a function that is called every time the value is asked for.

File: R/ospsuite-env.R [hunk 2]
```diff
@@ -60,5 +60,11 @@ getOSPSuiteSetting <- function(settingName){
     stop(messages$errorOSPSuiteSettingNotFound(settingName))
   }
 
-  return(ospsuiteEnv[[settingName]])
+  obj <- ospsuiteEnv[[settingName]]
```
Comment (PavelBal) [inline R/ospsuite-env.R:63]: A workaround to always return a value, and not a function.

Comment (msevestre) [reply R/ospsuite-env.R:63]: ah ah this is kind of badass cool

**Comment by msevestre** [comment]
Fix is missing for SensitivityAnalysisRunOptions. I'll update the PR

File: R/sensitivity-analysis-run-options.R [hunk 3]
```diff
@@ -13,17 +13,13 @@ SensitivityAnalysisRunOptions <- R6::R6Class(
     #' @param numberOfCores Number of cores to use for the simulation. Default value is \code{getOSPSuiteSetting("numberOfCores")}
     #' @param showProgress Should a progress information be displayed. Default value is \code{getOSPSuiteSetting("showProgress")}
     #' @return A new `SensitivityAnalysisRunOptions` object.
-    initialize = function(numberOfCores = ospsuiteEnv$numberOfCores,
-                              showProgress = ospsuiteEnv$showProgress) {
+    initialize = function(numberOfCores = NULL,
```
Comment (msevestre) [inline R/sensitivity-analysis-run-options.R:16]: This was missing here. Also I am differing initialization of number of cores  etc... to the function.
We had some issues in tlf or ospsuite::engine  and this also fixed some issues

**Pull request merged**